melodies_monet/driver.py

```diff
@@ -655,6 +860,10 @@ def mask_and_scale(self):
             for v in vars:
                 if v in self.variable_dict:
                     d = self.variable_dict[v]
+
+                    # array needs to be writeable 
+                    self.obj[v].data = self.obj[v].data.copy()
+                    
                     if 'unit_scale' in d:
```

Comment:
Let's take out this commit for now, but I made a note in case it comes up again.